implementation/src/main/java/io/smallrye/metrics/setup/config/MetricsConfigurationManager.java

```diff
@@ -0,0 +1,320 @@
+
+import io.smallrye.metrics.SharedMetricRegistries;
+import io.smallrye.metrics.setup.ApplicationNameResolver;
+
+public class MetricsConfigurationManager {
+
+    private static final String CLASS_NAME = MetricsConfigurationManager.class.getName();
+    private static final Logger LOGGER = Logger.getLogger(CLASS_NAME);
+
+    private ApplicationNameResolver anr = SharedMetricRegistries.getAppNameResolver();
+
+    static final String MP_PERCENTILES_PROP = "mp.metrics.distribution.percentiles";
+    static final String MP_HISTOGRAM_BUCKET_PROP = "mp.metrics.distribution.histogram.buckets";
+    static final String MP_TIMER_BUCKET_PROP = "mp.metrics.distribution.timer.buckets";
+    static final String MP_DEFAULT_BUCKET_PROP = "mp.metrics.distribution.percentiles-histogram.enabled";
+    static final String MP_HISTOGRAM_MAX_CONFIG = "mp.metrics.distribution.histogram.max-value";
+    static final String MP_HISTOGRAM_MIN_CONFIG = "mp.metrics.distribution.histogram.min-value";
+    static final String MP_TIMER_MAX_CONFIG = "mp.metrics.distribution.timer.max-value";
+    static final String MP_TIMER_MIN_CONFIG = "mp.metrics.distribution.timer.min-value";
+
+    private static MetricsConfigurationManager instance;
+
+    private MetricsConfigurationManager() {
+    };
+
+    private volatile Map<String, Collection<MetricPercentileConfiguration>> percentilesConfigMap = new HashMap<String, Collection<MetricPercentileConfiguration>>();
+
+    private volatile Map<String, Collection<HistogramBucketConfiguration>> histogramBucketsConfigMap = new HashMap<String, Collection<HistogramBucketConfiguration>>();;
+
+    private volatile Map<String, Collection<TimerBucketConfiguration>> timerBucketsConfigMap = new HashMap<String, Collection<TimerBucketConfiguration>>();
+
+    private volatile Map<String, Collection<DefaulBucketConfiguration>> defaultBucketConfigMap = new HashMap<String, Collection<DefaulBucketConfiguration>>();
+
+    private volatile Map<String, Collection<HistogramBucketMaxConfiguration>> defaultHistogramBucketMaxConfig = new HashMap<String, Collection<HistogramBucketMaxConfiguration>>();
+    private volatile Map<String, Collection<HistogramBucketMinConfiguration>> defaultHistogramBucketMinConfig = new HashMap<String, Collection<HistogramBucketMinConfiguration>>();
+
+    private volatile Map<String, Collection<TimerBucketMaxConfiguration>> defaultTimerBucketMaxConfig = new HashMap<String, Collection<TimerBucketMaxConfiguration>>();
+    private volatile Map<String, Collection<TimerBucketMinConfiguration>> defaultTimerBucketMinConfig = new HashMap<String, Collection<TimerBucketMinConfiguration>>();
+
+    public static synchronized MetricsConfigurationManager getInstance() {
+        if (instance == null) {
+            instance = new MetricsConfigurationManager();
+        }
+        return instance;
+    }
+
+    /**
+     * Returns the matching configuration object if it exists, null otherwise
```

Comment:
This sentence is used in 8 different methods.  This is not very helpful.

Reply:
updated, fixed.